When a new namespace is created in memory [Namespace#register()](https://github.com/lucmoreau/ProvToolbox/blob/ca05849c88a84aae736355569a8bdae78c68b072/modules-core/prov-model/src/main/java/org/openprovenance/prov/model/Namespace.java#L124) is used. This method does not validate the IRI.
Therefore, ProvToolbox allows documents with IRI with space to be created in memory. 

When serializing PROV-JSON format, there is no validation of IRI.
However, this is not wrong since the serializer should not expect ProvToolbox to generate a PROV document with an invalid IRI. The problem arises in the model where an invalid IRI can be generated. 

When deserializing from PROV-JSON format, Prov Python uses the [json](https://docs.python.org/3/library/json.html) library, more specifically, the json.load() function, which takes a custom decoder.
The custom decoder in [provjson](https://github.com/trungdong/prov/blob/master/src/prov/serializers/provjson.py) module is [ProvJSONDecoder](https://github.com/trungdong/prov/blob/master/src/prov/serializers/provjson.py#L101).
When deserializing the namespace, deserializer first reads the IRI, then in function [decode_json_container()](https://github.com/trungdong/prov/blob/1cb710a26c623c5c706b1c6945c645efd9e925c0/src/prov/serializers/provjson.py#L199), on line 204, just uses [Namespace constructor](https://github.com/trungdong/prov/blob/1cb710a26c623c5c706b1c6945c645efd9e925c0/src/prov/identifier.py#L85C9-L85C17) and [ProvBundle#add_namespace()](https://github.com/trungdong/prov/blob/1cb710a26c623c5c706b1c6945c645efd9e925c0/src/prov/model.py#L1335) which just calls [NamespaceManager#add_namespace()](https://github.com/trungdong/prov/blob/master/src/prov/model.py#L1080). There is no validation of IRI.

\`bundle.add_namespace(Namespace(prefix, uri))\`

Therefore, you can create a document with an invalid IRI in Prov Python.